project_wbs/models/account_analytic_account.py

```diff
@@ -0,0 +1,219 @@
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# Copyright 2017 Luxim d.o.o.
+# Copyright 2017 Matmoz d.o.o.
+# Copyright 2017 Deneroteam.
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class AccountAnalyticAccount(models.Model):
+    _inherit = 'account.analytic.account'
+    _order = 'complete_wbs_code'
+
+    @api.multi
+    def get_child_accounts(self):
+        result = {}
+        if not self.ids:
+            return result
+        for curr_id in self.ids:
+            result[curr_id] = True
+        # Now add the children
+        self.env.cr.execute('''
+        WITH RECURSIVE children AS (
+        SELECT parent_id, id
+        FROM account_analytic_account
+        WHERE parent_id IN %s
+        UNION ALL
+        SELECT a.parent_id, a.id
+        FROM account_analytic_account a
+        JOIN children b ON(a.parent_id = b.id)
+        )
+        SELECT * FROM children order by parent_id
+        ''', (tuple(self.ids),))
+        res = self.env.cr.fetchall()
+        for x, y in res:
+            result[y] = True
+        return result
+
+    @api.multi
+    def write(self, vals):
+        res = super(AccountAnalyticAccount, self).write(vals)
+        if vals.get('parent_id'):
+            for account in self.browse(self.get_child_accounts().keys()):
+                account._complete_wbs_code_calc()
+                account._complete_wbs_name_calc()
+        return res
+
+    @api.multi
+    @api.depends('code')
+    def _complete_wbs_code_calc(self):
+        for account in self:
+            data = []
+            acc = account
+            while acc:
+                if acc.code:
+                    data.insert(0, acc.code)
+
+                acc = acc.parent_id
+            if data:
+                if len(data) >= 2:
+                    data = '/'.join(data)
```

Comment:
I'm not sure this would be RTL-friendly, possibly use `reduce` with `_('%(l)s / %(r)s')` or something like that?